### ISSUE TITLE
When run without an explizit file given, ansible-lint does not run all checks

### ISSUE THREAD

**Issue by dbk-rabel** [ansible/ansible-lint#2736]
##### Summary

ansible-lint has an optional positional argument to name a file or path to check. It seems, that only providing explizit files works as expected.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
$ ansible-lint --version
ansible-lint 6.8.7 using ansible 2.14.0

```

- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

ping.yml:
```yaml (paste below)
- name: ping
  hosts: all
  tasks:
    - name: ping
      ansible.builtin.ping:
```

```console
$ ansible-lint --offline ping.yml
...
$ ansible-lint --offline .
...
$ ansible-lint --offline
...
```

(ping.yml is the only file in the folder.)

##### Desired Behavior

All three should report the same on ping.yml

##### Actual Behavior
```console
$ ansible-lint --offline ping.yml
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: ping.yml
WARNING  Listing 2 violation(s) that are fatal
name[casing]: All names should start with an uppercase letter. (warning)
ping.yml:1

name[casing]: All names should start with an uppercase letter. (warning)
ping.yml:4 Task/Handler: ping


         Rule Violation Summary
count tag          profile  rule associated tags
    2 name[casing] moderate idiom (warning)

Passed with basic profile, 1/5 star rating: 0 failure(s), 2 warning(s) on 1 files.

$ ansible-lint --offline .
Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.

$ ansible-lint --offline
Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
```

**Comment by piotrp** [comment]
This file isn't recognized as a playbook:
> Overriding detected file kind 'yaml' with 'playbook' for given positional argument: ping.yml

Default patterns (from https://github.com/ansible/ansible-lint/blob/v6.9.0/src/ansiblelint/config.py#L43) are:
```py
    {"playbook": "**/playbooks/*.{yml,yaml}"},
    {"playbook": "**/*playbook*.{yml,yaml}"},
    ...
    {"playbook": "**/molecule/*/*.{yaml,yml}"},  # molecule playbooks
```

You can override it using `kinds` setting: https://ansible-lint.readthedocs.io/configuring/#ansible-lint-configuration

**Comment by dbk-rabel** [comment]
Thanks for your answer!

I added this to my .ansible-lint :
```yaml
kinds:
  - playbook: ping.yml
```

This did not change the behaviour at all.

But even if this would work: Why does it make a difference if the file is given as a positional argument or not?
Because when we use `ansible-lint --offline ping.yml` it seems to notice that it is a playbook, even though it does not match the playbook pattern.

Yours
  David

**Comment by piotrp** [comment]
In my project, where all Ansible files are in "ansible" subdirectory, I'm using:
```
kinds:
  - playbook: "**/ansible/*.yml"
```

**Comment by dbk-rabel** [comment]
Thanks again. I am not exactly sure why, but  `playbook: "**/*.yml"` seems to do the trick for `ansible-lint --offline .`

That helps us to workaround that issue. Thank you!
